openhtf/__init__.py

```diff
@@ -213,13 +239,13 @@ def Execute(self, test_start=None, loop=None):
         raise InvalidTestStateError('Test already running', self._executor)
 
       # Snapshot some things we care about and store them.
-      self._test_data.metadata['test_name'] = self._test_options.name
-      self._test_data.metadata['config'] = conf._asdict()
+      self._test_desc.metadata['test_name'] = self._test_options.name
+      self._test_desc.metadata['config'] = conf._asdict()
       self.last_run_time_millis = util.TimeMillis()
 
       self._executor = exe.TestExecutor(
-          self._test_data, test_start, self._test_options.teardown_function)
-      _LOG.info('Executing test: %s', self.data.code_info.name)
+          self._test_desc, test_start, self._test_options.teardown_function)
+      _LOG.info('Executing test: %s', self.descriptor.code_info.name)
       self._executor.Start()
 
```

Comment:
At some point you moved output callbacks being called out of this class, but I think it should remain here. They're attached to the test and it makes sense that `Test` call them. If `executor.Finalize()` returned the final `test_state`, then this would call the callbacks and add to history. There's a log in there that mentions history, but it's the `Test` that adds the `history.append_record` callback in there, so it's really twisted as-is.